src/app/modules/library/select-all-items-checkbox/select-all-items-checkbox.component.ts

```diff
@@ -1,18 +1,36 @@
 import { Component, EventEmitter, Input, Output } from '@angular/core';
 import { MAT_CHECKBOX_DEFAULT_OPTIONS } from '@angular/material/checkbox';
 
+type SelectAllItemsStatus = 'none' | 'some' | 'all';
+
 @Component({
   selector: 'select-all-items-checkbox',
   templateUrl: './select-all-items-checkbox.component.html',
   providers: [{ provide: MAT_CHECKBOX_DEFAULT_OPTIONS, useValue: { clickAction: 'noop' } }]
 })
 export class SelectAllItemsCheckboxComponent {
   @Input() label: string = $localize`items`;
-  @Input() selectedAllItems: boolean = false;
-  @Input() selectedSomeItems: boolean = false;
-  @Output() selectAllItemsEvent: EventEmitter<boolean> = new EventEmitter<boolean>();
+  @Input() numAllItems: number;
+  @Input() numSelectedItems: number;
+  @Output() selectAllItemsEvent: EventEmitter<void> = new EventEmitter<void>();
+  @Output() selectNoItemsEvent: EventEmitter<void> = new EventEmitter<void>();
```

Comment:
Should selectNoItemsEvent be named unselectAllItemsEvent to be consistent and a little more accurate?

Reply:
As we discussed, I renamed these to "allSelectedEvent" and "noneSelectedEvent" to match class's status values.